docs/peeking.md

```diff
@@ -0,0 +1,19 @@
+## Peeking
+
+Peeking is implemented as per [MSC2753](https://github.com/matrix-org/matrix-doc/pull/2753).
+
+Implementationwise, this means:
+ * Users call `/peek` and `/unpeek` on the clientapi from a given device.
+ * The clientapi delegates these via HTTP to the roomserver, which coordinates peeking in general for a given room
+ * The roomserver writes an NewPeek event into the kafka log headed to the syncserver
+ * The syncserver tracks the existence of the local peek in its DB, and then starts waking up the peeking devices for the room in question, putting it in the `peek` section of the /sync response.
+
+Questions (given this is [my](https://github.com/ara4n) first time hacking on Dendrite):
+ * The whole clientapi -> roomserver -> syncapi flow to initiate a peek seems very indirect.  Is there a reason not to just let syncapi itself host the implementation of `/peek`?
```

Comment:
Requests to peek from the client and requests to peek from federation are the same/very similar, hence putting it in the roomserver makes more sense (clientapi -> roomserver and fedapi -> roomserver).

The one counterpoint is that this is truly only read-only: the room state is unmodified and the state stored is specific to syncapi only. We do make exceptions - filtering is a good example which exists in syncapi because it is specific to syncing only. If peeking didn't have the federation component I could argue that it should be in syncapi as it's just a filter (akin to the previous filter proposal), but the addition of federation complicates this.

Reply:
fedapi -> roomserver -> syncapi for the record will matter if the answer to https://github.com/matrix-org/dendrite/pull/1370#discussion_r481071577 is yes, as syncapi will need to inform users in the room of the members/counts of people peeking.

Reply:
yeah, having it in roomserver is making sense in the end.